dev/benchmarks/microbenchmarks/lib/ui/image_bench.dart

```diff
@@ -0,0 +1,148 @@
+  'packages/flutter_gallery_assets/products/platter.png',
+  'packages/flutter_gallery_assets/products/scarf.png',
+  'packages/flutter_gallery_assets/products/shirt.png',
+  'packages/flutter_gallery_assets/products/sunnies.png',
+  'packages/flutter_gallery_assets/products/sweater.png',
+  'packages/flutter_gallery_assets/products/sweats.png',
+  'packages/flutter_gallery_assets/products/table.png',
+  'packages/flutter_gallery_assets/products/teaset.png',
+  'packages/flutter_gallery_assets/products/top.png',
+  'packages/flutter_gallery_assets/people/square/ali.png',
+  'packages/flutter_gallery_assets/people/square/peter.png',
+  'packages/flutter_gallery_assets/people/square/sandra.png',
+  'packages/flutter_gallery_assets/people/square/stella.png',
+  'packages/flutter_gallery_assets/people/square/trevor.png',
+  'packages/shrine_images/diamond.png',
+  'packages/shrine_images/slanted_menu.png',
+  'packages/shrine_images/0-0.jpg',
+  'packages/shrine_images/1-0.jpg',
+  'packages/shrine_images/2-0.jpg',
+  'packages/shrine_images/3-0.jpg',
+  'packages/shrine_images/4-0.jpg',
+  'packages/shrine_images/5-0.jpg',
+  'packages/shrine_images/6-0.jpg',
+  'packages/shrine_images/7-0.jpg',
+  'packages/shrine_images/8-0.jpg',
+  'packages/shrine_images/9-0.jpg',
+  'packages/shrine_images/10-0.jpg',
+  'packages/shrine_images/11-0.jpg',
+  'packages/shrine_images/12-0.jpg',
+  'packages/shrine_images/13-0.jpg',
+  'packages/shrine_images/14-0.jpg',
+  'packages/shrine_images/15-0.jpg',
+  'packages/shrine_images/16-0.jpg',
+  'packages/shrine_images/17-0.jpg',
+  'packages/shrine_images/18-0.jpg',
+  'packages/shrine_images/19-0.jpg',
+  'packages/shrine_images/20-0.jpg',
+  'packages/shrine_images/21-0.jpg',
+  'packages/shrine_images/22-0.jpg',
+  'packages/shrine_images/23-0.jpg',
+  'packages/shrine_images/24-0.jpg',
+  'packages/shrine_images/25-0.jpg',
+  'packages/shrine_images/26-0.jpg',
+  'packages/shrine_images/27-0.jpg',
+  'packages/shrine_images/28-0.jpg',
+  'packages/shrine_images/29-0.jpg',
+  'packages/shrine_images/30-0.jpg',
+  'packages/shrine_images/31-0.jpg',
+  'packages/shrine_images/32-0.jpg',
+  'packages/shrine_images/33-0.jpg',
+  'packages/shrine_images/34-0.jpg',
+  'packages/shrine_images/35-0.jpg',
+  'packages/shrine_images/36-0.jpg',
+  'packages/shrine_images/37-0.jpg',
+];
+
+// Measures the time it takes to load a fixed number of assets into an
+// immutable buffer to later be decoded by skia.
+Future<void> main() async {
+  assert(false, "Don't run benchmarks in debug mode! Use 'flutter run --release'.");
+
+  final Stopwatch watch = Stopwatch();
+  await benchmarkWidgets((WidgetTester tester) async {
+    watch.start();
+    for (int i = 0; i < 10; i += 1) {
+      await Future.wait(<Future<ui.ImmutableBuffer>>[
+        for (String asset in assets)
+          rootBundle.load(asset).then((ByteData data) {
+            return ui.ImmutableBuffer.fromUint8List(data.buffer.asUint8List());
```

Comment:
My assumption is once the ImmutableBuffer.fromAsset change lands that'll be added here right?

I don't know if it makes sense to do here or to do in a separate benchmark, but WDYT of having something measuring the time it takes to `instantiateImageCodec` vs the new API?

Reply:
Yeah, i would just replace this code and we'd see the skia perf benchmark numbers change all at once.

> I don't know if it makes sense to do here or to do in a separate benchmark, but WDYT of having something measuring the time it takes to instantiateImageCodec vs the new API?

Assuming you already have the bytes loaded?

Reply:
I'm trying to suggest a measurement for getting the immutable buffer, and a measurement for getting an image codec from the buffer.

Your change will eliminate copies on both ends, right?

Reply:
Well, on one end. I think calling ui.ImmutableBuffer.fromUint8List includes the copy though. The codec doesn't have any more copies that we can eliminate IIRC

Reply:
ohhh I see now, you're already capturing that by calling `ImmutableBuffer.fromUint8List`. Ok.